mmda/types/annotation.py

```diff
@@ -2,18 +2,23 @@
 
 Annotations are objects that are 'aware' of the Document
 
-Collections of Annotations are how one constructs a new Iterable of Group-type objects within the Document
+Collections of Annotations are how one constructs a new
+Iterable of Group-type objects within the Document
 
 """
 from abc import abstractmethod
 from copy import deepcopy
 from dataclasses import dataclass, field
-from typing import Dict, Iterable, List, Optional
+from typing import TYPE_CHECKING, Dict, Iterable, List, Optional, Union
 from uuid import uuid4
 
 from mmda.types.box import Box
+from mmda.types.metadata import Metadata, store_field_in_metadata
 from mmda.types.span import Span
 
+if TYPE_CHECKING:
```

Comment:
wats happening here?

Reply:
You can't import `Document` from `mmda.types.document` during code execution because it will result into a circular import, but it's useful to have it around for static type checking. The solution is to use [`typing.TYPE_CHECKING`](https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING), which gets set to True by type checkers like `mypy`, but is false otherwise.

Reply:
didnt know that :O cool